### PR TITLE
Fix memory leak in groot and add trace logs

### LINKED ISSUE #2819: [BUG] The memory usage of Store in Groot will gradually increase

### PR DESCRIPTION
**Pull request by siyuan0322**
## What do these changes do?
1. Fix memory leak in groot
2. Fix compiler warnings
3. Bump up version of rocksdb and crossbeam-epoch
4. Add trace logs

## Related issue number

Fixes #2819

### CODE REVIEW

File: interactive_engine/executor/store/groot/src/db/storage/rocksdb.rs [hunk 1]
```diff
@@ -114,8 +114,9 @@ impl ExternalStorage for RocksDB {
     }
 
     fn open_backup_engine(&self, backup_path: &str) -> GraphResult<Box<dyn ExternalStorageBackup>> {
-        let backup_opts = BackupEngineOptions::default();
-        let backup_engine = BackupEngine::open(&backup_opts, backup_path).map_err(|e| {
+        let backup_opts = BackupEngineOptions::new(backup_path).unwrap();
```
Comment (BingqingLyu) [inline interactive_engine/executor/store/groot/src/db/storage/rocksdb.rs:117]: Please do not unwrap() directly. Instead,
```
let backup_opts =
            BackupEngineOptions::new(backup_path).map_err(|e| GraphError::Rocksdb(e.to_string()))?;
 let backup_engine = BackupEngine::open(
            &backup_opts,
            &rocksdb::Env::new().map_err(|e| GraphError::Rocksdb(e.to_string()))?,
        ).map_err(|e| {
         ...
        })?;
```

**Review by sighingnow — DISMISSED** [review]

File: interactive_engine/executor/assembly/groot/src/executor/gaia/engine_ports_response.rs [hunk 2]
```diff
@@ -49,7 +49,7 @@ impl Drop for EnginePortsResponse {
     fn drop(&mut self) {
         unsafe {
             if !self.errMsg.is_null() {
-                CString::from_raw(self.errMsg as *mut c_char);
+                drop(CString::from_raw(self.errMsg as *mut c_char));
```
Comment (longbinlai) [inline interactive_engine/executor/assembly/groot/src/executor/gaia/engine_ports_response.rs:52]: Can remove this unnecessary code.

Comment (siyuan0322) [reply interactive_engine/executor/assembly/groot/src/executor/gaia/engine_ports_response.rs:52]: This is for fixing the warnings of compiler.

File: interactive_engine/executor/assembly/groot/src/store/backup.rs [hunk 3]
```diff
@@ -44,13 +45,15 @@ pub extern "C" fn openGraphBackupEngine(
 #[no_mangle]
 pub extern "C" fn closeGraphBackupEngine(handle: GraphBackupHandle) {
     let ptr = handle as *mut GraphBackupEngine;
+    trace!("closeGraphBackupEngine");
     unsafe {
-        Box::from_raw(ptr);
+        drop(Box::from_raw(ptr));
```
Comment (longbinlai) [inline interactive_engine/executor/assembly/groot/src/store/backup.rs:50]: Can drop this unnecessary code

Comment (longbinlai) [reply interactive_engine/executor/assembly/groot/src/store/backup.rs:50]: Similarly, if a point already been wrapped into a Rust struct, e.g. Box, or CString, they will be dropped automatically

File: interactive_engine/executor/assembly/groot/src/store/jna_response.rs [hunk 4]
```diff
@@ -78,7 +78,7 @@ impl Drop for JnaResponse {
     fn drop(&mut self) {
         unsafe {
             if !self.errMsg.is_null() {
-                CString::from_raw(self.errMsg as *mut c_char);
+                drop(CString::from_raw(self.errMsg as *mut c_char));
```
Comment (longbinlai) [inline interactive_engine/executor/assembly/groot/src/store/jna_response.rs:81]: drop() not necessary.

File: interactive_engine/executor/assembly/groot/src/store/backup.rs (hunk 3 above)
Comment (sighingnow) [reply interactive_engine/executor/assembly/groot/src/store/backup.rs:50]: There would be a warning like

```
warning: unused return value of `std::boxed::Box::<T>::from_raw` that must be used
   --> /Users/runner/work/GraphScope/GraphScope/interactive_engine/executor/engine/pegasus/pegasus/src/communication/buffer.rs:121:13
    |
121 |             Box::from_raw(ptr.as_ptr());
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: call `drop(Box::from_raw(ptr))` if you intend to drop the `Box`
    = note: `#[warn(unused_must_use)]` on by default
help: use `let _ = ...` to ignore the resulting value
    |
121 |             let _ = Box::from_raw(ptr.as_ptr());
    |             +++++++

```

**Pull request merged**